_source/_posts/2022-06-14-simple-crud-react-and-spring-boot.md

```diff
@@ -0,0 +1,1348 @@
+        return ResponseEntity.ok().body(result);
+    }
+
+    @DeleteMapping("/group/{id}")
+    public ResponseEntity<?> deleteGroup(@PathVariable Long id) {
+        log.info("Request to delete group: {}", id);
+        groupRepository.deleteById(id);
+        return ResponseEntity.ok().build();
+    }
+}
+```
+
+Add the following dependency to your `pom.xml` to fix compilation errors:
+
+```xml
+<dependency>
+  <groupId>jakarta.validation</groupId>
+  <artifactId>jakarta.validation-api</artifactId>
+  <version>2.0.2</version>
+</dependency>
+```
+
+If you stop and start your server app and hit `http://localhost:8080/api/groups` with your browser, or a command line client, you should see the list of groups.
+
+You can create, read, update, and delete groups with the following [HTTPie](https://httpie.org) commands.
+
+```bash
+http POST :8080/api/group name='Utah JUG' city='Salt Lake City' country=USA
+http :8080/api/group/6
+http PUT :8080/api/group/6 id=6 name='Utah JUG' address='On the slopes'
+http DELETE :8080/api/group/6
+```
+
+<!--
+<build>
+  <defaultGoal>spring-boot:run</defaultGoal>
+  ...
+</build>
+-->
+
+## Create a React UI with Create React App
+
+Create React App is a command line utility that generates React projects for you. It's a convenient tool because it also offers commands that will build and optimize your project for production. It uses webpack under the covers to build everything.
+
+Create a new project in the `jugtours` directory with `npx`.
+
+```bash
+npx create-react-app@5 app
+```
+
+After the app creation process completes, navigate into the `app` directory and install [Bootstrap](https://getbootstrap.com/), cookie support for React, React Router, and [Reactstrap](https://reactstrap.github.io/).
+
+```bash
+cd app
+npm i bootstrap@5 react-cookie@4 react-router-dom@6 reactstrap@9
+```
+<!-- npm install bootstrap@5.2.0-beta1 fixes autprefixer issue -->
+
+You'll use Bootstrap's CSS and Reactstrap's components to make the UI look better, especially on mobile phones. If you'd like to learn more about Reactstrap, see [reactstrap.github.io](https://reactstrap.github.io). It has extensive documentation on its various components and how to use them.
+
+Add Bootstrap's CSS file as an import in `app/src/index.js`.
+
+```js
+import 'bootstrap/dist/css/bootstrap.min.css';
+```
+
+## Call your Spring Boot API and display the results
+
+Modify `app/src/App.js` to use the following code that calls `/api/groups` and display the list in the UI.
+
+```jsx
+import React, { useEffect, useState } from 'react';
+import logo from './logo.svg';
+import './App.css';
+
+const App = () => {
+
+  const [groups, setGroups] = useState([]);
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    setLoading(true);
+
+    fetch('api/groups')
+      .then(response => response.json())
+      .then(data => {
+        setGroups(data);
+        setLoading(false);
+      })
+  }, [setGroups, setLoading]);
```

Comment:
the array on a hook is to tell it to run only when the props in array changes so in this case I think you dont have to pass the state functions as I dont think they change
https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects

Reply:
I'll change it to `}, []);`. Please let me know if this is incorrect.